sierra_adapter/README.md

```diff
@@ -0,0 +1,288 @@
+    headers = {
+        "Authorization": f"Bearer {access_token}",
+        "Accept": "application/json",
+    }
+
+    sess.auth = HeadersAuth(headers=headers)
+
+    return sess
+
+
+
+if __name__ == "__main__":
+    sess = get_session(
+        oauth_key="<OAUTH_KEY>",
+        oauth_secret="<OAUTH_SECRET>"
+    )
+
+    resp = sess.get(API_URL + "/bibs", params={"id": "1645873"})
+    print(json.dumps(resp.json(), indent=2, sort_keys=True))
+```
+
+</details>
+
+The adapter accesses the Sierra API through a Scala library we've written: [sierra-streams-source](https://github.com/wellcometrust/sierra-streams-source).
+
+
+
+## How the adapter uses the Sierra API
+
+The Sierra API has over two million records.
+We don't want to fetch them all repeatedly – that would cause unnecessary load on the API servers, and needless churn (and cost) in the pipeline.
+
+Instead, we use the ability to request records which were updated within a particular timeframe.
+Every 15 minutes, we ask the API for a 30 minute *"window"* of updated records.
+
+<img style="width: 600px" src="images/sierra_windows.svg">
+
+This gives us a stream of updated records from the API.
+Because the windows overlap, we will eventually process every update to Sierra.
+
+If you need to get all the records, you can do so for asking for every window back to the installation of Sierra: request every update after **1 January 1999**.
+
+
+
+## How updates are scheduled
+
+Every 15 minutes, we want to poll the API for a 30 minute window of update records.
+How is that work scheduled?
+
+<img src="images/sierra_scheduler.svg">
+
+We run the following steps:
+
+1.  We have a timed CloudWatch event that fires every 15 minutes.
+    The event triggers a Lambda function (`sierra_window_generator`).
+
+2.  When the Lambda function wakes up, it gets the current time, and 30 minutes ago, and puts those times in a JSON blob:
+
+    ```json
+    {
+      "start": "2020-01-15T11:30:00+00:00",
+      "end":   "2020-01-15T12:00:00+00:00",
+    }
+    ```
+
+3.  The Lambda sends that JSON to an SNS topic of "windows that need to be processed", where it can be picked up by the rest of the adapter pipeline.
+
+We run two copies of this scheduler: one for bibs, one for items.
+
+Information about the windows that have been processed successfully is written into an S3 bucket (`wellcomecollection-platform-adapter-sierra`).
+If you look in `windows_bibs_complete` or `windows_items_complete`, you can see the windows that the adapter has successfully processed – and if there are any missing gaps.
+
+
+
+## How the adapter stores/processes data
+
+This is the architecture of the adapter:
+
+<img src="images/adapter_architecture.svg">
+
+We have the Sierra API on the left, and eventually we end up with a Sierra source data record in the VHS on the right.
+(The VHS is a data store used by the pipeline.)
+
+A "source data record" is a bib with all its associated items – the necessary pieces for a catalogue Work.
```

Comment:
Maybe this is all you need to say re `Work`s <> `Item`s? I think this is clearer than the above and communicates what's needed.